### ISSUE TITLE
Endless loop in POT/PO update regeneration

### ISSUE THREAD

**Issue by pedrobaeza** [OCA/oca-weblate-deployment#14]
Not sure if this is something of the Weblate components, or git bot, so I put it here and you tell me: in the repository OCA/social (https://github.com/OCA/social/commits/16.0/), there's an endless loop updating the POT, and the updating the PO, and redone again the POT, and so one:

![imagen](https://github.com/OCA/oca-weblate-deployment/assets/7165771/fb8aba74-010d-4069-890d-25a37f0ce690)

This is provoking a lot of noise, and undesired activity (we already needed a lot of work to bring commit history for the 17.0 version).

**Comment by sbidoul** [comment]
TL;DR: I think the bug is in Odoo.

.pot is generated with click-odoo-makepot after tests by the [GitHub workflow action](https://github.com/OCA/oca-addons-repo-template/blob/f94fc900b6d324a47151e3686836db70370c904f/src/.github/workflows/%7B%25%20if%20ci%20%3D%3D%20'GitHub'%20%25%7Dtest.yml%7B%25%20endif%20%25%7D.jinja#L102).

click-odoo-makepot uses the [standard odoo function to export .pot file](https://github.com/acsone/click-odoo-contrib/blob/8237607e4b63a544940b82ee0e3a077394cb97e4/click_odoo_contrib/makepot.py#L39-L42).

So in this case, it seems Odoo is not generating the same result in different runs of the .pot export on what should normally be the same database. Weblate aligns the .po with the .pot and pushes the changes which triggers a new build, etc.

**Comment by pedrobaeza** [comment]
@HydrionBurst maybe the order of the `#. odoo-javascript` comment is not deterministic? Check https://github.com/OCA/social/commit/82b8174698f584b1a101cddb2867b80fb3b0da37 for an example.

cc @chienandalu

**Comment by HydrionBurst** [comment]
> @HydrionBurst maybe the order of the `#. odoo-javascript` comment is not deterministic? Check [OCA/social@82b8174](https://github.com/OCA/social/commit/82b8174698f584b1a101cddb2867b80fb3b0da37) for an example.
> 
> cc @chienandalu

Oh, yes, it was a problem. I make a backport pr for 16.0
https://github.com/odoo/odoo/pull/172896

**Comment by pedrobaeza** [comment]
Thank you very much, Chong!

Closing this one then, as it will be sorted with the Odoo patch.
